chore: Upgrade to Tailwind v4

evadecker opened the pull request:
## What changed?
Upgrade the repo to Tailwind v4 and migrate to new CSS-based config. A few small style tweaks, like making the focus ring thicker. Resolves #441 

## Why?
This should make it easier to eventually migrate away from Tailwind if/when we choose to do that (#389), since more of the config is written using native CSS.

This enables us to get rid of some plugins like `tailwindcss/container-queries` which are now part of the core Tailwind package.

## How was this change made?
- Upgraded Tailwind to `@latest`
- Added `@tailwindcss/vite` package
- Removed `autoprefixer` and `postcss` packages, which are included by default in v4
- Removed `tailwindcss-animate`, which does not support v4, and replaced with `tw-animate-css`, which does, and has the same syntax
- Ran the Tailwind migration helper which automatically changed some classes
- Removed `tailwindcss-radix-colors`, which had some [compatibility issues with v4](https://github.com/mrcaidev/tailwindcss-radix-colors/issues/31)
- Install `@radix-ui/colors` directly
- Replace the vast majority of `dark:` Tailwind styles, since Radix automatically aliases all color variables for both light and dark mode, unlike the previous `tailwindcss-radix-colors` plugin, which required setting `color-gray-*` and `dark:color-graydark-*` everywhere.
- Disable CSS linting and formatting with Biome, which cannot parse Tailwind's language extensions such as `@plugin`, `@apply`, etc.

## How was this tested?
Clicked through all components in Storybook to verify styles look OK. Clicked through application. Tweaked anything that was out of the ordinary.

## Anything else?
🤞

evadecker commented:
Waiting to hear back about an issue with the Radix Colors Tailwind plugin: https://github.com/mrcaidev/tailwindcss-radix-colors/issues/31

Once that is resolved I can update this PR and mark it as ready for review.

evadecker commented:
@mbbischoff This is ready for review—don't feel obliged to do a line-by-line review of the code, but it would be helpful to open the app and Storybook and spot check components!

mbbischoff commented:
Unusre if this is a new issue or exists on main, but I ran into this error when testing the `Sections` storybook:

![CleanShot 2025-03-21 at 10 20 49@2x](https://github.com/user-attachments/assets/a2a375a0-6ed8-4817-92c5-c8233f538423)

mbbischoff commented:
Skeleton Circle renders the sekeleton as a rounded rectangle, is that the intended behavior?

![Uploading CleanShot 2025-03-21 at 10.28.13@2x.png…]()

mbbischoff commented:
Do we want our `Slider` to highlight the selected range in purple?

![CleanShot 2025-03-21 at 10 29 54@2x](https://github.com/user-attachments/assets/cff9e70e-3ea0-4c0f-8b52-766122e615a2)

mbbischoff commented:
In this `Toolbar` example the checkbox isn't vertically aligned 
![CleanShot 2025-03-21 at 10 34 07@2x](https://github.com/user-attachments/assets/6969bf91-6344-40ec-982f-1924a661237c)

mbbischoff commented on src/styles/index.css:
```diff
@@ -178,3 +178,19 @@ body,
+
+.animate-focus-ring-in {
+  animation: animate-focus-ring-in 300ms ease forwards;
+}
```
Perhaps toss a newline at EOF for easier merges later.

mbbischoff reviewed (approved):
This looks really good! Approving with a few minor comments, which probably exist on `main` so not blocking here.

evadecker commented:
> Do we want our `Slider` to highlight the selected range in purple?

Updated and added a single thumb example:

![CleanShot 2025-03-21 at 11 11 29@2x](https://github.com/user-attachments/assets/0d2c1dfa-7db5-4b18-9b60-660f53e9ce8e)
![CleanShot 2025-03-21 at 11 11 44@2x](https://github.com/user-attachments/assets/e6cd8140-b075-4af4-8fb6-8344e7e8c7eb)

evadecker commented:
> Skeleton Circle renders the sekeleton as a rounded rectangle, is that the intended behavior?

This works fine in the app, I think for some reason Storybook is failing to parse / generate Tailwind `classNames` when they are passed in as an argument. Not sure why, filing to return to later: #445

evadecker commented:
> Unusre if this is a new issue or exists on main, but I ran into this error when testing the `Sections` storybook:
> 
> ![CleanShot 2025-03-21 at 10 20 49@2x](https://private-user-images.githubusercontent.com/14093/425504780-a2a375a0-6ed8-4817-92c5-c8233f538423.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI1NzA0MzMsIm5iZiI6MTc0MjU3MDEzMywicGF0aCI6Ii8xNDA5My80MjU1MDQ3ODAtYTJhMzc1YTAtNmVkOC00ODE3LTkyYzUtYzgyMzNmNTM4NDIzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzIxVDE1MTUzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ0Y2YyNjdjMjhhMzc1OTY4ZjA5MTY0NDE5MzEwOGNkOTYwN2RkNmUxNDBjNDAzY2IzZWZlMzk4YmYwYTMyODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.KsRHfdnn-l99AaCD0C7OkDkJ3k8-_8VOM3dQC5rooSo)

Which "Sections"?

![CleanShot 2025-03-21 at 11 20 11@2x](https://github.com/user-attachments/assets/9e76c5f5-77f2-4355-b025-b9ac4a8f60d5)

I'm not getting this error on any of mine.

mbbischoff commented:
> > Do we want our `Slider` to highlight the selected range in purple?
> 
> Updated and added a single thumb example:
> 
> ![CleanShot 2025-03-21 at 11 11 29@2x](https://private-user-images.githubusercontent.com/4117920/425529490-0d2c1dfa-7db5-4b18-9b60-660f53e9ce8e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI1NzA3NDksIm5iZiI6MTc0MjU3MDQ0OSwicGF0aCI6Ii80MTE3OTIwLzQyNTUyOTQ5MC0wZDJjMWRmYS03ZGI1LTRiMTgtOWI2MC02NjBmNTNlOWNlOGUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMjFUMTUyMDQ5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWJmOTE0MmE1NjFkMTdiMjliZmFjMzgzNzQ3NzNkMWRkYzVlNDQzMGE4YWRjZDQwMDg0Y2ZiMWI1MWM3MzQ0ZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.aTeZLFXxReNj5xow8POC4H1M_2bdVcxborlmFsJtwj4) ![CleanShot 2025-03-21 at 11 11 44@2x](https://private-user-images.githubusercontent.com/4117920/425529506-e6cd8140-b075-4af4-8fb6-8344e7e8c7eb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI1NzA3NDksIm5iZiI6MTc0MjU3MDQ0OSwicGF0aCI6Ii80MTE3OTIwLzQyNTUyOTUwNi1lNmNkODE0MC1iMDc1LTRhZjQtOGZiNi04MzQ0ZTdlOGM3ZWIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMjFUMTUyMDQ5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGM0MWM3NjEyMTNkZjhhMDUxY2QxOWVjYmM2MTA2MGUzMGZjMDMxMTdmZjYxOTU2MjAxMWU4NjRkYWE5MDljMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.6iggfpvSNKsf79fRE0eAwIL9j0CK-PyyXhwj28asv8c)

goregous!

mbbischoff commented:
> > Unusre if this is a new issue or exists on main, but I ran into this error when testing the `Sections` storybook:
> > ![CleanShot 2025-03-21 at 10 20 49@2x](https://private-user-images.githubusercontent.com/14093/425504780-a2a375a0-6ed8-4817-92c5-c8233f538423.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI1NzA0MzMsIm5iZiI6MTc0MjU3MDEzMywicGF0aCI6Ii8xNDA5My80MjU1MDQ3ODAtYTJhMzc1YTAtNmVkOC00ODE3LTkyYzUtYzgyMzNmNTM4NDIzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzIxVDE1MTUzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ0Y2YyNjdjMjhhMzc1OTY4ZjA5MTY0NDE5MzEwOGNkOTYwN2RkNmUxNDBjNDAzY2IzZWZlMzk4YmYwYTMyODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.KsRHfdnn-l99AaCD0C7OkDkJ3k8-_8VOM3dQC5rooSo)
> 
> Which "Sections"?
> 
> ![CleanShot 2025-03-21 at 11 20 11@2x](https://private-user-images.githubusercontent.com/4117920/425533101-9e76c5f5-77f2-4355-b025-b9ac4a8f60d5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI1NzA3NDksIm5iZiI6MTc0MjU3MDQ0OSwicGF0aCI6Ii80MTE3OTIwLzQyNTUzMzEwMS05ZTc2YzVmNS03N2YyLTQzNTUtYjAyNS1iOWFjNGE4ZjYwZDUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMjFUMTUyMDQ5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDAzNzJjYjc1MWI3M2JlODUzZTk1NzMzNGFlODFmMzRkYzllMDRkZGNiNzJmZTRlODA4ZWRkMTk5OTJhZmNmMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.jMkFSLcUPUmHI2zCv9lMgbLFBeEis0UkbpXZGmNLnqs)
> 
> I'm not getting this error on any of mine.

ComboBox.Sections when clicking into the box
